src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -2184,210 +1647,237 @@ public static T GetElement<T>(this Vector128<T> vector, int index)
-
-            for (int index = 0; index < Vector128<T>.Count; index++)
-            {
-                T value = Scalar<T>.LessThan(left.GetElementUnsafe(index), right.GetElementUnsafe(index)) ? Scalar<T>.AllBitsSet : default;
-                result.SetElementUnsafe(index, value);
-            }
-
-            return result;
+            return Create(
+                Vector64.LessThan(left._lower, right._lower),
+                Vector64.LessThan(left._upper, right._upper)
+            );
         }
 
         /// <summary>Compares two vectors to determine if all elements are less.</summary>
+        /// <typeparam name="T">The type of the elements in the vector.</typeparam>
         /// <param name="left">The vector to compare with <paramref name="right" />.</param>
         /// <param name="right">The vector to compare with <paramref name="left" />.</param>
-        /// <typeparam name="T">The type of the elements in the vector.</typeparam>
         /// <returns><c>true</c> if all elements in <paramref name="left" /> were less than the corresponding element in <paramref name="right" />.</returns>
+        /// <exception cref="NotSupportedException">The type of <paramref name="left" /> and <paramref name="right" /> (<typeparamref name="T" />) is not supported.</exception>
         [Intrinsic]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool LessThanAll<T>(Vector128<T> left, Vector128<T> right)
-            where T : struct => LessThan(left, right).As<T, ulong>() == Vector128<ulong>.AllBitsSet;
+            where T : struct
+        {
+            return Vector64.LessThanAll(left._lower, right._lower)
+                && Vector64.LessThanAll(left._upper, right._upper);
+        }
 
         /// <summary>Compares two vectors to determine if any elements are less.</summary>
+        /// <typeparam name="T">The type of the elements in the vector.</typeparam>
         /// <param name="left">The vector to compare with <paramref name="right" />.</param>
         /// <param name="right">The vector to compare with <paramref name="left" />.</param>
-        /// <typeparam name="T">The type of the elements in the vector.</typeparam>
         /// <returns><c>true</c> if any elements in <paramref name="left" /> was less than the corresponding element in <paramref name="right" />.</returns>
+        /// <exception cref="NotSupportedException">The type of <paramref name="left" /> and <paramref name="right" /> (<typeparamref name="T" />) is not supported.</exception>
         [Intrinsic]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool LessThanAny<T>(Vector128<T> left, Vector128<T> right)
-            where T : struct => LessThan(left, right).As<T, ulong>() != Vector128<ulong>.Zero;
+            where T : struct
+        {
+            return Vector64.LessThanAny(left._lower, right._lower)
+                && Vector64.LessThanAny(left._upper, right._upper);
```

Comment:
This shouldn't be || instead of && ?

Reply:
Does this indicate a hole in unit test coverage?

Reply:
No, the tests were failing and did catch this.